src/Backend/Modules/MediaGalleries/Domain/MediaGallery/MediaGallery.php

```diff
@@ -0,0 +1,403 @@
+        $this->status = $status;
+        $this->text = $text;
+    }
+
+    /**
+     * To array
+     *
+     * @return array
+     */
+    public function __toArray(): array
+    {
+        return [
+            'id' => $this->id,
+            'userId' => $this->userId,
+            'moduleExtraId' => $this->moduleExtraId,
+            'action' => $this->action,
+            'title' => $this->title,
+            'text' => $this->text,
+            'createdOn' => $this->createdOn->getTimestamp(),
+            'editedOn' => $this->editedOn->getTimestamp(),
+            'publishOn' => $this->publishOn->getTimestamp(),
+            'status' => (string) $this->status,
+            'mediaGroup' => $this->mediaGroup->__toArray(),
+        ];
+    }
+
+    /**
+     * Gets the value of id.
+     *
+     * @return string
+     */
+    public function getId(): string
+    {
+        return $this->id;
+    }
+
+    /**
+     * Gets the value of userId.
+     *
+     * @return integer
+     */
+    public function getUserId(): int
+    {
+        return $this->userId;
+    }
+
+    /**
+     * Gets the value of moduleExtraId.
+     *
+     * @return integer
+     */
+    public function getModuleExtraId(): int
+    {
+        return $this->moduleExtraId;
+    }
+
+    /**
+     * Gets the Action
+     *
+     * @return string
+     */
+    public function getAction(): string
+    {
+        return $this->action;
+    }
+
+    /**
+     * Gets the value of title.
+     *
+     * @return string
+     */
+    public function getTitle(): string
+    {
+        return $this->title;
+    }
+
+    /**
+     * Gets the value of text.
+     *
+     * @return string|null
+     */
+    public function getText()
+    {
+        return $this->text;
+    }
+
+    /**
+     * Gets the value of createdOn.
+     *
+     * @return \DateTime
+     */
+    public function getCreatedOn(): \DateTime
+    {
+        return $this->createdOn;
+    }
+
+    /**
+     * Gets the value of editedOn.
+     *
+     * @return \DateTime
+     */
+    public function getEditedOn(): \DateTime
+    {
+        return $this->editedOn;
+    }
+
+    /**
+     * Gets the value of publishOn.
+     *
+     * @return \DateTime
+     */
+    public function getPublishOn(): \DateTime
+    {
+        return $this->publishOn;
+    }
+
+    /**
+     * @return Status
+     */
+    public function getStatus(): Status
+    {
+        return $this->status;
+    }
+
+    /**
+     * Gets the value of mediaGroup.
+     *
+     * @return MediaGroup
+     */
+    public function getMediaGroup(): MediaGroup
+    {
+        return $this->mediaGroup;
+    }
+
+    /**
+     * @ORM\PrePersist
+     */
+    public function onPrePersist()
+    {
+        $this->createdOn = $this->editedOn = new \Datetime();
+
+        $this->moduleExtraId = Model::insertExtra(
+            ModuleExtraType::widget(),
+            'MediaGalleries',
+            'Gallery',
+            'Gallery',
+            array(),
```

Comment:
fork 5 is going to switch to the short array syntax (#2001) maybe also use it in your module already